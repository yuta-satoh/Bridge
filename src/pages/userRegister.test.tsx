import { render, screen } from "@testing-library/react";
import UserEvent from "@testing-library/user-event";

import UserRegister from "./userRegister";

jest.mock('next/router', () => ({
    useRouter() {
      return {
        replace: jest.fn(),
        push: jest.fn(),
        reload: jest.fn(),
      };
    },
}));

const fetchMockOk = () => {
    return new Promise((resolve) => {
        resolve({
            ok: true,
            status: 200,
            statusText: 'OK',
        })
    })
}

const fetchMockNg = () => {
    return new Promise((resolve) => {
        resolve({
            ok: false,
            status: 400,
            statusText: 'NG',
        })
    })
}

describe('会員登録ページ', () => {
    test('正常に描画', () => {
        const { asFragment } = render(<UserRegister />);
        expect(asFragment()).toMatchSnapshot();
    })
})
