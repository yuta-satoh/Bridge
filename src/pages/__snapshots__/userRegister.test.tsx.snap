// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`会員登録ページ 正常に描画 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    <div
      class="urbody"
    >
      <div
        class="title"
      >
        <h1
          class="h1"
        >
          JOIN TO Bridge
        </h1>
        <p
          class="p"
        >
          Bridge会員に登録する
        </p>
      </div>
      <form
        action="/api/register"
        class="form"
        method="post"
      >
        <div
          class="inputItems"
        >
          <label
            class="label"
            for="lastName"
          >
            お名前
          </label>
          <span
            class="primary"
          >
            必須
          </span>
          <br />
          <input
            class="input border border-neutral-500 rounded pl-2.5"
            id="lastName"
            name="lastName"
            placeholder="例：山田"
            type="text"
            value=""
          />
          <input
            class="input border border-neutral-500 rounded pl-2.5"
            id="firstName"
            name="firstName"
            placeholder="例：花子"
            type="text"
            value=""
          />
          <p
            class="error"
          />
        </div>
        <div
          class="inputItems"
        >
          <label
            class="label"
            for="female"
          >
            性別
          </label>
          <span
            class="primary"
          >
            必須
          </span>
          <br />
          <div
            class="radioField"
          >
            <div
              class="radioForm border border-neutral-500 rounded pl-2.5"
            >
              <input
                class="radioButton"
                id="female"
                name="gender"
                type="radio"
                value="female"
              />
              <label
                class="genderLabel"
                for="female"
              >
                女性 
              </label>
            </div>
            <div
              class="radioForm border border-neutral-500 rounded pl-2.5"
            >
              <input
                class="radioButton"
                id="male"
                name="gender"
                type="radio"
                value="male"
              />
              <label
                class="genderLabel"
                for="male"
              >
                男性 
              </label>
            </div>
            <div
              class="radioForm border border-neutral-500 rounded pl-2.5"
            >
              <input
                class="radioButton"
                id="other"
                name="gender"
                type="radio"
                value="other"
              />
              <label
                class="genderLabel"
                for="other"
              >
                その他
              </label>
            </div>
          </div>
          <p
            class="error"
          />
        </div>
        <div
          class="inputItems"
        >
          <label
            class="label"
            for="email"
          >
            メールアドレス
          </label>
          <span
            class="primary"
          >
            必須
          </span>
          <br />
          <input
            class="inputParts border border-neutral-500 rounded pl-2.5"
            id="email"
            name="email"
            placeholder="例：bridge@example.com"
            type="email"
            value=""
          />
          <p
            class="error"
          />
        </div>
        <div
          class="inputItems"
        >
          <label
            class="label"
            for="tell"
          >
            電話番号
          </label>
          <span
            class="primary"
          >
            必須
          </span>
          <br />
          <input
            class="number border border-neutral-500 rounded pl-2.5"
            id="tell1"
            name="tell1"
            placeholder="xxx"
            type="text"
            value=""
          />
          <span>
            -
          </span>
          <input
            class="number border border-neutral-500 rounded pl-2.5"
            id="tell2"
            name="tell2"
            placeholder="xxxx"
            type="text"
            value=""
          />
          <span>
            -
          </span>
          <input
            class="number border border-neutral-500 rounded pl-2.5"
            id="tell3"
            name="tell3"
            placeholder="xxxx"
            type="text"
            value=""
          />
          <p
            class="error"
          />
        </div>
        <div
          class="inputItems"
        >
          <label
            class="label"
            for="zipcode"
          >
            郵便番号
          </label>
          <span
            class="primary"
          >
            必須
          </span>
          <br />
          <input
            class="number border border-neutral-500 rounded pl-2.5"
            id="zipcode1"
            name="zipcode1"
            pattern="^[0-9]+$"
            placeholder="xxx"
            type="text"
            value=""
          />
          <span>
            -
          </span>
          <input
            class="number border border-neutral-500 rounded pl-2.5"
            id="zipcode2"
            name="zipcode2"
            pattern="^[0-9]+$"
            placeholder="xxxx"
            type="text"
            value=""
          />
          <button
            class="zipButton text-white bg-neutral-900 border border-neutral-900 rounded px-1"
            type="button"
          >
            住所検索
          </button>
          <p
            class="error"
          />
        </div>
        <div
          class="inputItems"
        >
          <label
            class="label"
            for="address"
          >
            住所
          </label>
          <span
            class="primary"
          >
            必須
          </span>
          <br />
          <input
            class="inputParts border border-neutral-500 rounded pl-2.5"
            id="address"
            name="address"
            placeholder="例：東京都新宿区新宿2丁目x-x-x"
            type="text"
            value=""
          />
          <p
            class="error"
          />
        </div>
        <div
          class="inputItems"
        >
          <label
            class="label"
            for="password"
          >
            パスワード
          </label>
          <span
            class="primary"
          >
            必須
          </span>
          <br />
          <input
            class="inputParts border border-neutral-500 rounded pl-2.5"
            data-testid="password"
            id="password"
            name="password"
            placeholder="例：abcdef123456"
            type="password"
            value=""
          />
          <p
            class="error"
          />
        </div>
        <div
          class="inputItems"
        >
          <label
            class="label"
            for="confirmationPassword"
          >
            パスワード確認用
          </label>
          <span
            class="primary"
          >
            必須
          </span>
          <br />
          <input
            class="inputParts border border-neutral-500 rounded pl-2.5"
            data-testid="confirmationPassword"
            id="confirmationPassword"
            name="confirmationPassword"
            placeholder="例：abcdef123456"
            type="password"
            value=""
          />
          <p
            class="error"
          />
          <p
            class="p"
          >
            ※確認のためパスワードを再入力して下さい
          </p>
          <br />
        </div>
        <div
          class="buttonArea"
        >
          <button
            class="submitButton"
            type="submit"
          >
            会員登録をする
            <span
              class="buttonSpan"
            >
              →
            </span>
          </button>
        </div>
        <div
          class="buttonArea"
        >
          <button
            class="clearButton"
            type="button"
          >
            入力クリア
          </button>
        </div>
      </form>
      <div
        class="title"
      >
        <h2
          class="subTitle"
        >
          会員の方はこちら
        </h2>
      </div>
      <div
        class="loginLink"
      >
        <a
          href="/login"
        >
          <button
            class="linkButton"
            type="button"
          >
            ログイン
            <span
              class="buttonSpan"
            >
              →
            </span>
          </button>
        </a>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
